### PR TITLE
sql: use a custom parser for the SQL compiler

### PR DESCRIPTION
**Pull request by lalithsuresh**
We currently use calcite-server's DDL parser, but would like to extend this with functionality from calcite-babel.  The only way to compose parsers like this is to create a custom parser by using FMPP/FTL files to customize Calcite's parser template.

This patch adds a custom parser that replaces the currently used DdlParser, and sets it up for further babel extensions.

At build time, we extract Parser.jj from the calcite jar, combine it with our FMPP/FTL files, and run java-cc on the result to codegen our custom parser.

### CODE REVIEW

File: sql-to-dbsp-compiler/SQL-compiler/src/main/codegen/includes/ddl.ftl [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        }
+    |
+        { list.add(id); }
+    )
+|
+    id = SimpleIdentifier() {
+        list.add(id);
+    }
+|
+    [ <CONSTRAINT> { s.add(this); } name = SimpleIdentifier() ]
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/codegen/includes/ddl.ftl:78]: I wasn't planning to support constraints, I am not sure what to do about them.

File: sql-to-dbsp-compiler/SQL-compiler/src/main/codegen/includes/ddl.ftl [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+SqlCreate SqlCreateTable(Span s, boolean replace) :
+{
+    final boolean ifNotExists;
+    final SqlIdentifier id;
+    SqlNodeList tableElementList = null;
+    SqlNode query = null;
+}
+{
+    <TABLE> ifNotExists = IfNotExistsOpt() id = CompoundIdentifier()
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/codegen/includes/ddl.ftl:162]: I am not sure that "if exists" makes sense in our context
where is the CREATE keyword?

**Review by mihaibudiu — APPROVED** [review]

File: sql-to-dbsp-compiler/SQL-compiler/src/main/codegen/includes/ddl.ftl (hunk 2 above)
Comment (lalithsuresh) [reply sql-to-dbsp-compiler/SQL-compiler/src/main/codegen/includes/ddl.ftl:162]: If people are copy pasting DDL from their databases, it's fine to let it in?

File: sql-to-dbsp-compiler/SQL-compiler/src/main/codegen/includes/ddl.ftl (hunk 1 above)
Comment (lalithsuresh) [reply sql-to-dbsp-compiler/SQL-compiler/src/main/codegen/includes/ddl.ftl]: Okay, I've removed it for now. I'll keep UNIQUE and PRIMARY KEY.

**Pull request merged**